distribution/registry.go

```diff
@@ -60,14 +60,17 @@ func NewV2Repository(ctx context.Context, repoInfo *registry.RepositoryInfo, end
 	endpointStr := strings.TrimRight(endpoint.URL, "/") + "/v2/"
 	req, err := http.NewRequest("GET", endpointStr, nil)
 	if err != nil {
-		return nil, false, err
+		return nil, false, false, err
```

Comment:
Is there anyway we can embed this in the error type? I am reticent in expanding the number of return values from this function.

Let me know if you need ideas.